packages/react-components/react-combobox/src/components/Option/Option.types.ts

```diff
@@ -6,6 +6,12 @@ export type OptionSlots = {
 
   /* The check icon that is visible for selected options */
   checkIcon: Slot<'span'>;
+
+  /**
+   * Option children are placed in this slot
+   * Avoid using the `children` property on this slot in favor of direct Option children.
+   */
+  content: NonNullable<Slot<'span'>>;
```

Comment:
Summarizing an offline chat: I think we can get away without adding the extra slot, if you add the content padding to the root, and use margins on the icon slot to get the appropriate padding/gap for the icon.

Reply:
done!